### PR TITLE
[WIP] Batfishify

### PR DESCRIPTION
**Pull request by davidtheclark**
@lshig is leading this effort, and I'm helping out. Opening this PR now for visibility.

We're starting fresh and gradually pulling in existing files and code bits, so this PR is going to include a lot of Git changes. Looking at the diff probably isn't as informative as pulling the branch and browsing.

I'm thinking that once @lshig is done, @cammace and I can review the code and content (comparing this branch to the live production site to check against copy-paste mistakes). @lshig and I can work with @cammace to make sure you understand how everything works.

Update from @lshig: @davidtheclark @jfurrow are definitely leading this effort now. 👏

### CODE REVIEW

**Comment by davidtheclark** [comment]
@jfurrow I noticed you've been adding Flow types. Flow is not included in this build system, though — it's only in www (and Studio), for now. Let's use PropTypes, instead, and talk about whether we want to extend the reach of Flow.

**Comment by jfurrow** [comment]
@davidtheclark Thanks for catching that; it's leftover relics from my copy/pasting from `master`. I'll remove!

File: src/pages/map-sdk/examples/index.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ let MapSdkExamplesLayout = class MapSdkExamplesLayout extends React.Component {
       <PageShell
         frontMatter={{
           title: 'Android Map SDK Examples',
-          link: 'foo',
+          link: '/map-sdk/examples',
           rank: 1,
```
Comment (davidtheclark) [inline src/pages/map-sdk/examples/index.js]: Link and rank are no longer necessary.

Comment (jfurrow) [inline src/pages/map-sdk/examples/index.js]: Gone!

**Comment by davidtheclark** [comment]
Things are moving along. All of the Markdown documentation has been ported into Batfish pages. Examples and tutorials are almost there. Here's a little checklist of things I know still need to be done. Please chime in if you think of more items!

- [ ] Finish examples and tutorials.
- [ ] System for publishing JavaDocs at the right URLs.
- [ ] Clean up headers. — @cammace 
- [ ] Write meta descriptions. — @cammace 
- [ ] Sort out design questions (e.g. colors).
- [ ] Make sure we've incorporated any changes since this branch was created.
- [ ] Cross-browser testing. I'll copy the checklist from www.
  - [ ] Check on mobile-size viewport.
  - [ ] Check on tablet-size viewport.
  - [ ] Check on laptop-size viewport.
  - [ ] Check on big-monitor-size viewport.
  - [ ] Check in Chrome.
  - [ ] Check in Firefox.
  - [ ] Check in Safari.
  - [ ] Check in Edge.
  - [ ] Check in IE11.
  - [ ] Check in mobile Safari.
  - [ ] Check in Android's Chrome.

**Comment by jfurrow** [comment]
When testing this in IE11, React doesn't render anything and I'm seeing the following cryptic error:
screenshot 2017-08-07 10 45 20

After some Googling and reading [this issue in the React repo](https://github.com/facebook/react/issues/6472), I think there are two versions of React being loaded. `npm ls react` shows the following:
```
├─┬ @mapbox/batfish@0.9.0
│ └─┬ @mapbox/babel-plugin-transform-jsxtreme-markdown@0.4.0
│   └─┬ @mapbox/jsxtreme-markdown@0.7.1
│     └─┬ htmltojsx@0.2.6
│       └── react@0.14.9
└── react@15.6.1
```

@davidtheclark I'll keep investigating, but I wondered if you know of a quick solution, or know where to look next?

**Comment by davidtheclark** [comment]
Hm — there's no reason the Babel plugin, jsxtreme-markdown or htmltojsx should be included in the browser bundle. If they are, it's a bug. 

It may be that Batfish's function for creating the client-side bundle needs to specifically look for peer dependencies react and react-dom in the application's node_modules directory, if that's possible?

**Pull request merged**